internal/summary/summary.go

```diff
@@ -29,9 +31,10 @@ type k8sObject struct {
 }
 
 // NewSummary returns summary state
-func NewSummary() *Summary {
+func NewSummary(client k8s.Client) *Summary {
```

Comment:
i'm a little bit nervous about giving Summary a k8sClient versus making it the responsibility of the caller to pass this data in. wdyt?

Reply:
on the one hand, it would be nice if the function was "pure" and didn't go look things up from external systems. on the other hand, if our long-term plan is to show a HUD of your cluster state, maybe this is inevitable

Reply:
:-/ i'm also bummed about this... is it super ugly if you expect the caller to pass the data?

Reply:
ok, i tried something different where i pass in a resolver function, which feels less icky. wdyt?